mewarpx/tests/test_emission.py

```diff
@@ -672,3 +670,69 @@ def test_plasma_injector_fixedT2():
             df[label_base + key + '_std'] = np.std(val)
 
     assert testing_util.test_df_vs_ref(testname=name, df=df, margin=0.3)
+
+
+def test_arbitrary_distribution_emitter():
+    name = "arbitrary_distribution_emitter"
+    # Include a random run number to allow parallel runs to not collide.  Using
+    # python randint prevents collisions due to numpy rseed below
+    testing_util.initialize_testingdir(name)
+
+    # Initialize each run with consistent, randomly-chosen, rseed. Use a random
+    # seed instead for initial dataframe generation.
+    # np.random.seed(11874889)
+
```

Comment:
```suggestion

NX = NZ = 512
```

Then use NX and NZ instead of hardcoded 512 throughout.